sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -67,6 +67,11 @@
         ('example', _INPUT_TYPE),
         ('inference', _OUTPUT_TYPE),
     ])
+PredictionResult.__doc__ = """A NamedTuple containing both input and output
+  from the inference."""
+PredictionResult.example.__doc__ = """The input example."""
+PredictionResult.inference.__doc__ = """Results for the inference on the model
+  for the given example."""
```

Comment:
I guess I missed this. Why aren't we using the class form of NamedTuple? Is it because the type variables/generics don't work?